### PR TITLE
Add spec v2

### PR DESCRIPTION
**Pull request by aftersnow**

### CODE REVIEW

File: specs-go/v2/architecture.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+//
+// Supported features:
+// - Attention mechanisms: Multi-Head Attention (MHA) and Grouped Query Attention (GQA)
+// - Activation functions: GELU, ReLU
+// - Position embeddings: Rotary Position Embedding (RoPE)
+// - Normalization: RMSNorm (Root Mean Square Layer Normalization)
+//
+// This structure is designed to be flexible and accommodate various transformer architectures
+// used in state-of-the-art language models.
+type TransformerForCausalLLM struct {
```
Comment (bergwolf) [inline specs-go/v2/architecture.go:14]: This is not referenced anywhere. Let's add it when config.go references it?

File: specs-go/v2/mediatype.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	MediaTypeModelManifest = "application/vnd.cnai.model.manifest.v2+json"
+)
+
+// configs
+const (
+	// MediaTypeModelConfig specifies the media type for model configuration.
+	MediaTypeModelConfig = "application/vnd.cnai.model.config.v2+json"
+
+	// MediaTypeModelLicense specifies the media type for model license.
+	MediaTypeModelLicense = "application/vnd.cnai.model.license.v2+json"
```
Comment (bergwolf) [inline specs-go/v2/mediatype.go:15]: Why is the license file in json format? Looks like `plaintext` is better.

File: specs-go/v2/mediatype.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+	// MediaTypeModelExtension specifies the media type for model configuration extension.
+	MediaTypeModelExtension = "application/vnd.cnai.model.extension.v2+json"
+)
+
+// processors
+const (
+	// MediaTypeModelProcessorText specifies the media type for text processors.
+	// This includes tokenizers like sentencepiece, used for processing textual input.
+	MediaTypeModelProcessorText = "application/vnd.cnai.model.processor.text.v2.tar"
```
Comment (bergwolf) [inline specs-go/v2/mediatype.go:28]: The processors are source code files right? Let's compress it with `tar+gzip` or `tar+zstd`?

Comment (bergwolf) [reply specs-go/v2/mediatype.go:28]: ok, I see that zstd is still wip.

File: specs-go/v2/architecture.go (hunk 1 above)
Comment (aftersnow) [reply specs-go/v2/architecture.go:14]: ok

File: specs-go/v2/mediatype.go (hunk 2 above)
Comment (aftersnow) [reply specs-go/v2/mediatype.go:15]: yes

**Pull request merged**